kv/memberlist/tcp_transport.go

```diff
@@ -91,7 +99,11 @@ type TCPTransport struct {
 	tcpListeners []net.Listener
 	tlsConfig    *tls.Config
 
-	shutdown atomic.Int32
+	writeCh chan writeRequest
+	writeWG sync.WaitGroup
+
+	shutdown   bool
+	shutdownMu sync.RWMutex
```

Comment:
nit: mutex is typically on the top of the group of variables it protects. ("mutex hat")